BuildScript.Nuke/BaseBuild.Compile.cs

```diff
@@ -66,49 +73,63 @@ public partial class BaseBuild : NukeBuild
-        .DisableRestore()
-        .When(GitRepository != null, s => s.SetPackageProjectUrl(GitRepository!.HttpsUrl))
-        .CombineWith(projectFiles, (settings, projectFile) =>
-            settings.SetToolsVersion(projectFile.ToolsVersion)
-                .SetTargetPath(projectFile.ProjectPath)
-                .SetTargets(GetCompileTargets(projectFile))
-                .SetConfiguration(projectFile.Configuration)
-                .SetInformationalVersion(SemanticVersion.GetAssemblyInformationalVersion(projectFile.Configuration, AdditionalBuildMetadata))
-        )
-    );
+    var projectFilesDebugTargets = projectFiles
+        .Where(file => file.Configuration == "Debug").ToList();
+    var projectFilesReleaseTargets = projectFiles
+        .Where(file => file.Configuration == "Release").ToList();
+    if (projectFilesDebugTargets.Count > 0) CompileProject(projectFilesDebugTargets, "Debug");
+
+    if (projectFilesReleaseTargets.Count > 0) CompileProject(projectFilesReleaseTargets, "Release");
+  }
+
+  private void CompileProject (IReadOnlyCollection<ProjectMetadata> projects, string configuration)
+  {
+    var sortedProjects = SortedProjectList;
+    var filteredProjects = sortedProjects.Select(projectSort => projects.SingleOrDefault(project => projectSort.FilePath == project.ProjectPath)).WhereNotNull().Distinct();
+
+    filteredProjects.ForEach(project =>
+    {
+      MSBuild(s => s
+          .SetProcessToolPath(ToolPath.Value)
+          .SetAssemblyVersion(SemanticVersion.AssemblyVersion)
+          .SetFileVersion(SemanticVersion.AssemblyFileVersion)
+          .SetCopyright(AssemblyMetadata.Copyright)
+          .SetProperty(MSBuildProperties.PackageVersion, SemanticVersion.AssemblyNuGetVersion)
+          .SetProperty(MSBuildProperties.CompanyName, AssemblyMetadata.CompanyName)
+          .SetProperty(MSBuildProperties.CompanyUrl, AssemblyMetadata.CompanyUrl)
+          .SetProperty(MSBuildProperties.ProductName, AssemblyMetadata.ProductName)
+          .SetProperty(MSBuildProperties.AssemblyOriginatorKeyFile, Directories.SolutionKeyFile)
+          .When(GitRepository != null, s => s.SetPackageProjectUrl(GitRepository!.HttpsUrl))
+          .SetToolsVersion(project.ToolsVersion)
+          .SetConfiguration(configuration)
+          .SetInformationalVersion(
+              SemanticVersion.GetAssemblyInformationalVersion(configuration, AdditionalBuildMetadata))
+          .SetTargetPath(project.ProjectPath)
+          .SetTargets(project.IsMultiTargetFramework ? MSBuildTargets.DispatchToInnerBuilds : MSBuildTargets.Build)
+      );
+    });
   }
 
   private string GetToolPath ()
   {
-    var toolPath = MSBuildTasks.MSBuildPath;
+    var toolPath = MSBuildPath;
     var editions = new[] { "Enterprise", "Professional", "Community", "BuildTools", "Preview" };
     if (!string.IsNullOrEmpty(MsBuildPath))
       toolPath = MsBuildPath;
```

Comment:
you could just return the `MsBuildPath` here, or return the result of the statement in the `else if` block. Anyway, what if no property is set? Then the returned `toolPath` would be null. I think it would be OK to `Assert.Fail()` if nothing is available. Additionally, the code should also work if only VS2022 Professional is installed, but in this case, the installation would not be found because the non-X86 folder is not checked. Please create a follow up issue for this.